File: src/test/cpp/model/filetest.cpp
// Copyright 2013-2021 Phokham Nonava
//
// Use of this source code is governed by the MIT license that can be
// found in the LICENSE file.

#include "model/file.h"

#include "gtest/gtest.h"

using namespace pulse;

TEST(filetest, testValues) {
	for (auto file: file::values) {
		EXPECT_EQ(file, file::values[file]);
	}
}

File: pulse-java/src/test/java/com/fluxchess/pulse/java/model/FileTest.java
/*
 * Copyright 2013-2023 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */
package com.fluxchess.pulse.java.model;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class FileTest {

	@Test
	void testValues() {
		for (int file : File.values) {
			assertThat(File.values[file]).isEqualTo(file);
		}
	}
}